### PR TITLE
Navigation component: Infer back button label from parent menu title

### LINKED ISSUE #25247: Navigation Component: Improve the Back Navigation
- Explore if it's possible to automatize the back button label for nested menus.
Right now consumers have to provide the label manually (or it defaults to "Back").

https://github.com/WordPress/gutenberg/blob/668a710b3eb9c942b21eb5913fbe304b15ea71dd/packages/components/src/navigation/menu.js#L41-L50 

By traversing the `Navigation` children we should be able to infer the parent menu's title and automatically use it as back button label for its children menus.

Depends on #25246 exploration.

---

- Add an `onBackButtonClick` handler.

### PR DESCRIPTION
**Pull request by david-szabo97**
Fixes: https://github.com/WordPress/gutenberg/issues/25247
Depends on: https://github.com/WordPress/gutenberg/pull/25340

## Description
Right now we need to manually specific `backButtonLabel` or it defaults to `Back`. Rather than always doing it manually, we can infer it from the parent menu.

If `backButtonLabel` is not specified, then the parent menu's title will be the label of the back button. If for some reason parent menu's title is not available we still fallback to `Back`.

## How has this been tested?
`yarn storybook:dev`

## Types of changes
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by Copons** [comment]
This looks good to me! ✨ 
Let's keep it on hold until #25340 is merged!

**Comment by david-szabo97** [comment]
#25340 is merged. Rebased, ready for an approve!

**Review by Copons — APPROVED** [review]

**Pull request merged**
